Tests/vrc-oscquery-tests/MeaModDiscoveryTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Net;
+
+namespace VRC.OSCQuery.Tests;
+
+public class MeaModDiscoveryTests
+{
+    private const int DELAY_TIME_MS = 1000;
+    
+    [Test]
+    public async Task Discovery_WhenOSCQueryServiceAdvertised_ContainsService()
+    {
+        // Set up Advertising and Discovery
+        var advertiser = new MeaModDiscovery();
+        var discoverer = new MeaModDiscovery();
+        var oscQueryServiceProfile = new OSCQueryServiceProfile
+        (
+            Guid.NewGuid().ToString(),
+            IPAddress.Loopback,
+            Extensions.GetAvailableTcpPort(),
+            OSCQueryServiceProfile.ServiceType.OSCQuery
+        );
+
+        // Advertise the service, Refresh the services, and wait for the service to be discovered
+        advertiser.Advertise(oscQueryServiceProfile);
+        await Task.Delay(DELAY_TIME_MS); // Wait for the service to be advertised
+        discoverer.RefreshServices();
+        await Task.Delay(DELAY_TIME_MS); // Wait for the service to be discovered
```

Comment:
can we not use the other side of MDNS here and just wait for a service to appear in the callback? We could kick it off as 2 tasks one which simply resolves after X maximum test timeout (e.g. 5 seconds) and the other as soon as the service is found?

Instead of just waiting a flat amount

Reply:
I tried that approach, making a TaskCancellationSource which resolves once the right service is found, or times out if it's not. It worked once and failed 5 times, whereas this approach worked every time for me. I don't love this approach, though.

I looked up the actual [MeaMod Tests](https://github.com/meamod/MeaMod.DNS/tree/master/test) at the end of this weekend programming session - too late for another refactor on that day but probably worth investigating to see how they test things.